ir/state.cpp

```diff
@@ -95,14 +123,22 @@ const StateValue& State::getAndAddUndefs(const Value &val) {
   return v;
 }
 
+const StateValue& State::getAndAddPoisonUB(const Value &val) {
+  analysis.non_poison_vals.insert(&val);
+  auto &np = get<1>(values[values_map.at(&val)]).first.non_poison;
```

Comment:
this is not correct because np may refer undef variables.

Reply:
Sorry, fixed it. It was tricky, hope it is correct this time.

Reply:
looks good to me, thanks!
BTW, can you run tests again after these changes to check if the results are the same?

Reply:
Sure. I'll leave the LLVM unit test results first, then the single file benchmarks.